pkg/operator/ceph/file/controller.go

```diff
@@ -225,7 +225,19 @@ func (c *FilesystemController) filesystemOwner(fs *cephv1.CephFilesystem) metav1
 }
 
 func filesystemChanged(oldFS, newFS cephv1.FilesystemSpec) bool {
-	if len(oldFS.DataPools) != len(newFS.DataPools) {
+	if oldFS.MetadataPool.Replicated.Size != newFS.MetadataPool.Replicated.Size {
+		logger.Infof("number of MetaDataPool changed from %d to %d", oldFS.MetadataPool.Replicated.Size, newFS.MetadataPool.Replicated.Size)
+		return true
+	}
+
+	if len(oldFS.DataPools) == len(newFS.DataPools) {
+		for i := 0; i < len(oldFS.DataPools); i++ {
+			if oldFS.DataPools[i].Replicated.Size != newFS.DataPools[i].Replicated.Size {
+				logger.Infof("existing DataPools replicated size changed from %d to %d", oldFS.DataPools[i].Replicated.Size, newFS.DataPools[i].Replicated.Size)
```

Comment:
This is done in case multiple  `dataPools` have  been added and now the operator can detect exactly which dataPool's replicated size has been changed.
Thanks for the tip @ashishranjan738

Reply:
We should return true in this case